packages/core/src/Site/index.js

```diff
@@ -651,17 +658,23 @@ class Site {
 
     try {
       await this.readSiteConfig(baseUrl);
-      this.collectAddressablePages();
-      await this.collectBaseUrl();
-      this.collectUserDefinedVariablesMap();
-      await this.buildAssets();
-      await (this.onePagePath ? this.lazyBuildSourceFiles() : this.buildSourceFiles());
-      await this.copyCoreWebAsset();
-      await this.copyBootstrapTheme(false);
-      await this.copyFontAwesomeAsset();
-      await this.copyOcticonsAsset();
-      await this.copyMaterialIconsAsset();
-      await this.writeSiteData();
+      this.ignoreVersionFiles('');
+      await this.buildSiteHelper();
```

Comment:
A slight problem with building the main site first before copying the versions will result in the live preview refreshing multiple times.

https://user-images.githubusercontent.com/47470981/164440392-34914826-9ca5-4559-8d84-9d0509fe1e67.mp4

Steps to replicate:
1. Have about 5-6 versions so that it becomes obvious
2. `markbind serve -v`

I believe shifting `this.buildSiteHelper` to after the call to `this.addVersions` should resolve the issue.

Reply:
I'm having difficulty replicating this : ( I'll try it on a different computer, but I seem to get the same behaviour with both orders of this line, with 5 versions of the markbind docs website (which I think should be big enough(?). I've moved it for now - if you or anyone else has time I'd appreciate if you could check if that fixes the issue!

Reply:
> I've moved it for now - if you or anyone else has time I'd appreciate if you could check if that fixes the issue!

Yup this seems to have fixed the issue :)